teuthology/provision/maas.py

```diff
@@ -0,0 +1,426 @@
+        )
+
+        if resp.text == "Machine is locked":
+            log.info("Machine %s is locked", self.shortname)
+        elif data := resp.json():
+            if not data.get("locked"):
+                raise RuntimeError(
+                    "Machine %s locking failed, status: %s",
+                    self.shortname, data.get("locked")
+                )
+
+    def unlock_machine(self) -> None:
+        """Unlock the machine"""
+        resp = self.do_request(
+            f"/machines/{self.system_id}/op-unlock", method="POST"
+        )
+
+        if resp.text == "Cannot unlock an already-unlocked node":
+            log.info("Machine %s is not locked", self.shortname)
+        elif data := resp.json():
+            if data.get("locked"):
+                raise RuntimeError(
+                    "Machine %s locking failed, status %s",
+                    self.shortname, data.get("locked")
+                )
+
+    def deploy_machine(self) -> None:
+        """Deploy the machine"""
+        image_data: Dict[str, Any] = self.get_image_data()
+        if image_data.get("type", "").lower() not in ["synced", "uploaded"]:
+            raise RuntimeError(
+                "MaaS image %s is not synced, current status: %s",
+                image_data.get("name"), image_data.get("type")
+            )
+
+        log.info(
+            "Deploying machine %s with image %s", self.shortname, image_data.get("name")
+        )
+        files = {
+            "distro_series": (None, image_data.get("name")),
+            "user_data": (None, self._get_user_data()),
+        }
+        data: Dict[str, Any] = self.do_request(
+            f"/machines/{self.system_id}/op-deploy", method="POST", files=files
+        ).json()
+        if data.get("status_name", "").lower() != "deploying":
+            raise RuntimeError(
+                "Machine %s deployment failed, status: %s",
+                self.shortname, data.get("status_name")
+            )
+
+    def release_machine(self, erase: bool = True) -> None:
+        """Release the machine
+
+        :param erase: Optional parameter to indicate whether to erase disks
+                        (default: True)
+        """
+        data: Dict[str, bool] = {"erase": erase}
+        resp: Dict[str, Any] = self.do_request(
+            f"/machines/{self.system_id}/op-release", method="POST", data=data
+        ).json()
+        if resp.get("status_name", "").lower() not in ["disk erasing", "releasing"]:
+            raise RuntimeError(
+                "Machine %s releasing failed, status %s",
+                self.shortname, resp.get("status_name")
+            )
+
+    def abort_deploy(self) -> None:
+        """Abort deployment of the machine"""
+        machine = self.get_machines_data()
+        _status = machine.get("status_name", "").lower()
+        if _status != "deploying":
+            log.info("Unexpected status %s to abort operation", _status)
```

Comment:
```suggestion
            self.log.info(f"Unexpected status '{_status}'. Aborting deploy operation")
```
I'm unsure if the wording I'm suggesting here is 100% accurate, but the quoting helps make the message more readable.